### PR TITLE
Trigger global operation notification during solution close

### LINKED ISSUE #47408: Solution Crawler should suspend operations during solution close
In https://github.com/dotnet/roslyn/pull/47359, we suspended turning project references -> binary references during solution close. 

@sharwell had a further suggestion:

> For me, this change drops solution close from ~12 seconds to ~8 seconds. Significant work is still done by custom COM handling in the running document table. If Roslyn wanted to make further changes, we could suspend the Solution Crawler as soon as we get a Solution Closing notification.

This bug is representing that suggestion.

### PR DESCRIPTION
**Pull request by mavasani**
Fixes #47408

Fixes [AB#1414916](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1414916): `[Responsiveness] IncrementalAnalyzerProcessor+NormalPriorityProcessor.GetPrioritizedPendingDocuments consumes large amounts of CPU during solution close (24%/2400ms of devenv CPU)`

We currently trigger a global operation during solution open and build. These global operations pause background analysis, such as solution crawler. Dave found a performance spike in the solution crawler during solution close. We don't see a realistic reason to keep running the solution crawler while the solution is closing, hence we now pause it during solution close.

### CODE REVIEW

File: src/VisualStudio/Core/Def/SolutionEventMonitor.cs [hunk 1]
```diff
@@ -62,6 +72,9 @@ public void Dispose()
                 _notificationService = null;
                 KnownUIContexts.SolutionBuildingContext.UIContextChanged -= SolutionBuildingContextChanged;
                 KnownUIContexts.SolutionOpeningContext.UIContextChanged -= SolutionOpeningContextChanged;
+
+                if (_solutionClosingContext != null)
+                    _solutionClosingContext.UIContextChanged -= SolutionClosingContextChanged;
```
Comment (sharwell) [inline src/VisualStudio/Core/Def/SolutionEventMonitor.cs:77]: 📝 This null check is unnecessary/inconsistent

```suggestion
                _solutionClosingContext.UIContextChanged -= SolutionClosingContextChanged;
```

File: src/VisualStudio/Core/Def/SolutionEventMonitor.cs [hunk 2]
```diff
@@ -45,6 +45,16 @@ public SolutionEventMonitor(VisualStudioWorkspace workspace)
                 }
 
                 KnownUIContexts.SolutionOpeningContext.UIContextChanged += SolutionOpeningContextChanged;
+
+                _solutionClosingContext = UIContext.FromUIContextGuid(VSConstants.UICONTEXT.SolutionClosing_guid);
```
Comment (sharwell) [inline src/VisualStudio/Core/Def/SolutionEventMonitor.cs:49]: 💡 Move this initialization outside the `if` statement

File: src/VisualStudio/Core/Def/SolutionEventMonitor.cs [hunk 3]
```diff
@@ -12,15 +10,17 @@
...
+    /// The intent is to suspend analysis of non-essential files for the duration of these solution operations.
     /// </summary>
     internal sealed class SolutionEventMonitor : IDisposable
     {
         private const string SolutionBuilding = "Solution Building";
         private const string SolutionOpening = "Solution Opening";
+        private const string SolutionClosing = "Solution Closing";
 
-        private IGlobalOperationNotificationService _notificationService;
+        private readonly UIContext? _solutionClosingContext;
```
Comment (sharwell) [inline src/VisualStudio/Core/Def/SolutionEventMonitor.cs:22]:
```suggestion
        private readonly UIContext _solutionClosingContext;
```

**Review by jasonmalinowski — APPROVED** [review]

File: src/VisualStudio/Core/Def/SolutionEventMonitor.cs (hunk 2 above)
Comment (mavasani) [reply src/VisualStudio/Core/Def/SolutionEventMonitor.cs:49]: @sharwell Entire functionality within the `SolutionEventMonitor` type is conditioned on the below if check inside its constructor: https://github.com/dotnet/roslyn/blob/fa9926dbc9d6b6677f96efc5b47e72037562ae1b/src/VisualStudio/Core/Def/SolutionEventMonitor.cs#L28

I am just adding to existing patterns for solution open and build contexts, so am hesitant to introduce inconsistency in the code on how solution close is dealt as compared to open/build.

Comment (mavasani) [reply src/VisualStudio/Core/Def/SolutionEventMonitor.cs:49]: @sharwell If you still feel strongly about this, a separate PR cleaning up this entire type and possibly removing the below check would be more suitable: https://github.com/dotnet/roslyn/blob/fa9926dbc9d6b6677f96efc5b47e72037562ae1b/src/VisualStudio/Core/Def/SolutionEventMonitor.cs#L28

Comment (mavasani) [reply src/VisualStudio/Core/Def/SolutionEventMonitor.cs:49]: Filed https://github.com/dotnet/roslyn/issues/58827 if we want to cleanup this type.

**Pull request merged**